File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/Account.fs
namespace Back.Types.DataBase

open System
open System.ComponentModel.DataAnnotations

type Account(name: string, email: string, passwordHash: string) =
    class
        [<Key>]
        member val Id = 0L with get

        member val Name = name with get, set
        member val Email = email with get, set
        member val PasswordHash = passwordHash with get, set
        member val EmailConfirmed = false with get, set
        member val Locked = false with get, set
        member val BannedTo = DateTime.MinValue with get, set
        member val LastGrobalMessageReadTime = DateTime.MinValue with get, set

        member val ThisHourPostSend = 0 with get, set
        member val ThisDayPostSend = 0 with get, set
        member val ThisHourPostCommentSend = 0 with get, set
        member val ThisDayPostCommentSend = 0 with get, set
        member val ThisHourTreehollowSend = 0 with get, set
        member val ThisDayTreehollowSend = 0 with get, set
        member val ThisHourTreehollowCommentSend = 0 with get, set
        member val ThisDayTreehollowCommentSend = 0 with get, set
        member val ThisDayUnresivedPrivateMessageSend = 0 with get, set
        member val LastConfirmEmailSendTime = DateTime.Now with get, set
    end

File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/Account.cs
﻿using System.ComponentModel.DataAnnotations;

namespace Back.Types.DataBase;

public class Account
{
    [Key]
    public long Id { get; set; }

    public string Name { get; set; }
    public string Email { get; set; }
    public byte[] PasswordHash { get; set; }
    public bool EmailConfired { get; set; } = false;
    public bool IsAdmin { get; set; } = false;
    public bool Locked { get; set; } = false;
    public DateTime BannedTo { get; set; } = DateTime.MinValue;
    public DateTime LastGlobalMessageReadTime { get; set; } = DateTime.MinValue;

    public int ThisHourPostSend { get; set; } = 0;
    public int ThisDayPostSend { get; set; } = 0;
    public int ThisHourPostCommentSend { get; set; } = 0;
    public int ThisDayPostCommentSend { get; set; } = 0;
    public int ThisHourTreehollowSend { get; set; } = 0;
    public int ThisDayTreehollowSend { get; set; } = 0;
    public int ThisHourTreehollowCommentSend { get; set; } = 0;
    public int ThisDayTreehollowCommentSend { get; set; } = 0;
    public int ThisDayUnresivedPrivateMessageSend { get; set; } = 0;
    public DateTime LastConfirmEmailSendTime { get; set; } = DateTime.MinValue;

    public Account(string name, string email, byte[] passwordHash)
    {
        Name = name;
        Email = email;
        PasswordHash = passwordHash;
    }

    public Account()
    {

    }
}
